lib/playlistGenerator/ConcatSource.js

```diff
@@ -120,39 +120,40 @@ ConcatSource.prototype.getItem = function(index) {
 
 var popClip = function() {
     var that = this;
-    removeItem.call(that,0);
+    removeItems.call(that, 0, 1);
 };
 
 var allKeysButDurations = _.filter(Object.keys(ConcatSource.prototype.dataScheme),function(key){
     return key !== 'durations' && key !== 'lastEncoderDTS';
 });
 
-var removeItem = function(index,surpressEvent) {
+// removeItems input parameters:
+//      index: index of 1st item to remove
+//      count: total number of items to remove
+//      supressEvent: if 'true' fire event per removed clip
+var removeItems = function(index, count=1,  surpressEvent=false) {
     var that = this;
 
-    that.logger.debug('Remove Item(%j) %j', index, that.getItem(index));
+    that.logger.debug('Remove Item(%j) %j count=%d', index, that.getItem(index), count);
 
     if(!surpressEvent) {
-        that.emit(playlistUtils.ClipEvents.clip_removed, that, index);
+        that.emit(playlistUtils.ClipEvents.clip_removed, that, index, count);
     }
 
-    that.durationsMan.remove(index);
-    that.keyFrameDurations.splice(index,1);
+    for (let i=0; i<count; i++) {
 
-    if(index === 0) {
-        _.each(allKeysButDurations,function(key){
-            this[key].shift();
-        }, that.inner);
-    } else {
-        _.each(allKeysButDurations,function(key){
-            this[key].splice(index,1);
-        }, that.inner);
+        that.durationsMan.remove(index);
```

Comment:
this change was reverted. durationMan is an object. Therefor splice is not supported. Can consider in future to add splice method to TimestampList.